src/FileSystem/SourceFileFilter.php

```diff
@@ -81,15 +92,21 @@ public function getFilters(): array
      */
     public function filter(iterable $input): iterable
     {
-        if ($this->filters === []) {
-            return $input;
+        $iterator = $this->iterableToIterator($input);
+
+        if ($this->filters !== []) {
+            $iterator = new RealPathFilterIterator(
+                $iterator,
+                $this->filters,
+                []
+            );
+        }
+
+        if ($this->excludeDirectories !== []) {
+            $iterator = new PathFilterIterator($iterator, [], $this->excludeDirectories);
```

Comment:
is this right? So if you have both `filters` and `excludes`, you and up with ignoring filters here

Reply:
why?

Reply:
ah I see, nevermind then